cranelift/codegen/src/machinst/lower.rs

```diff
@@ -465,6 +466,15 @@ impl<'func, I: VCodeInst> Lower<'func, I> {
                 for insn in self.vcode.abi().gen_copy_arg_to_regs(i, regs).into_iter() {
                     self.emit(insn);
                 }
+                if self.abi().signature().params[i].purpose == ArgumentPurpose::StructReturn {
+                    assert!(regs.len() == 1);
+                    let ty = self.abi().signature().params[i].value_type;
+                    self.emit(I::gen_move(
+                        Writable::from_reg(self.retval_regs[0].regs()[0]),
```

Comment:
Can you add a comment referring to the `legalize_signature` function (or whatever is its new name), saying that it ensured its first `retval` register *is* the struct return value?

Also... would this unnecessarily add spill pressure for the entire function's lifetime to do it here (since one more register is blocked), instead of doing in the retval setup function?

Reply:
Good point; actually it could be *not* the first return value, so I've changed this to be more general and just find the StructReturn one (panic'ing if not present).

Re: register pressure, yes, possibly, though we're basically passing a value through from an arg to a return value so we either have to keep it in a vreg (note that `retval_regs` is filled with vregs; the move to rax happens in the epilogue) or create a pseudo-spill-slot ourselves to save it; we can't recompute it later. So I don't think we can do better than this given the ABI, sadly...

Reply:
> Good point; actually it could be not the first return value, so I've changed this to be more general and just find the StructReturn one (panic'ing if not present).

Is there a test for this? (and if there's not, can you add one please?)